File: rides-service/src/main/java/com/modsen/ride/client/PassengerServiceClientFallback.java
package com.modsen.ride.client;

import com.modsen.ride.util.ExceptionMessages;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
@Slf4j
public class PassengerServiceClientFallback implements PassengerServiceClient {
    @Override
    public ResponseEntity<Boolean> doesPassengerExists(UUID passengerId) {
        throw new RuntimeException(ExceptionMessages.UNABLE_TO_REACH_PASSENGER_SERVICE.format());
    }
}

File: rating-service/src/main/kotlin/com/modsen/rating/client/PassengerServiceClientFallback.kt
package com.modsen.rating.client

import com.modsen.rating.util.ExceptionMessages
import java.util.UUID
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Component

@Component
class PassengerServiceClientFallback: PassengerServiceClient {
    override fun doesPassengerExist(passengerId: UUID): ResponseEntity<Boolean> {
        throw RuntimeException(ExceptionMessages.UNABLE_TO_REACH_PASSENGER_SERVICE)
    }
}